### DISCUSSION TITLE
How can I use modelx-cython module?

### DISCUSSION THREAD

**Discussion by happyych** [fumitoh/modelx#98] [Q&A]
Hello,

I am thrilled while reading the post about modelx-cython module?
After I read the post, I really want to try modelx-cython module to check the speed improvement.
However, I have some trouble during the test.

1) I write  "sample.py" and "spec.py" ( I guess you may change the name "spec" to "config" since you commented as "config.py" in github )

2) I checked the library installation for 5 modules
    modelx v0.23.0+, Cython v3.0.0+, setuptools, libcst, MonkeyType

3) install modelx-cython module   using the code you gave  "pip install modelx-cython"
    and check that  all the modules are installed  (using pip list)  -> version 0.0.4

4) finally, I try the code you showed,   " mx2cy **model** "
   however,  there is an error code  
    mx2cy m_ex
          ^
SyntaxError: invalid syntax

also, when I tried to import the modelx-cython module,   I faced another error code

    import modelx-cython
                 ^
SyntaxError: invalid syntax.

Could you recommend for this issue?

Thank you!

**Comment by fumitoh** [comment]
It seems like you attempted to execute `mx2cy m_ex` as if it were Python code. However, `mx2cy` is not a Python statement but a command-line tool, intended to be used from the Windows Command Prompt. To run it correctly, you should open the Command Prompt and enter the command there. You can type `mx2cy -h` in the Command Prompt to access the help menu and verify that the tool is installed correctly.

Please be aware that `modelx-cython` is currently in a highly experimental phase. So far, it has only been tested with the `BasicTerm_S` model in `lifelib`. As a result, you might encounter errors when using it with other models. Despite this, we  appreciate error reports as they are invaluable for identifying issues and improve `modelx-cython`.

**Reply by happyych** [reply]
Thank you for the explanation. As you let me know, I entered the code in Command Prompt. 
I guess modelx-cython library is successfully installed since I can access the help menu through mx2cy -h.
However, I faced another problem. 
the error messege is "ValueError: 'BasicTerm_S_cy\_mx_sys.py' doesn't match any files"
Could you advise me how can I solve this problem? 
Thanks!
<img width="733" alt="ERROR1" src="https://github.com/fumitoh/modelx/assets/82881835/ab6111df-4e5d-4d82-ba99-45373a59fa3b">

<img width="362" alt="ERROR2" src="https://github.com/fumitoh/modelx/assets/82881835/c4c215c2-5a53-450b-91b4-2e3145e631a6">

**Reply by fumitoh** [reply]
You need to pass an exported model to `mx2cy`.

Below is a Python script to create an exported model named `BasicTerm_S_nomx` from `BasicTerm_S` in the current folder.

```python
import modelx as mx

mx.read_model("BasicTerm_S").export("BasicTerm_S_nomx")
```

Then you can run 

`mx2cy BasicTerm_S_nomx`

from the Windows Command Prompt.

**Reply by happyych** [reply]
Thank you for the fast reply!  I finally success to cythonize the model. I will try mx2cy on my own model either.
By the way, I want to ask you the way to run the model.
Is there any other way to run a lot of "id"(modelpoint) rather than using dynamic space as you show me on the sample code?
It might be the problem of my laptop, but I think raising bunch of dynamic space cause some memory problem.
I want to check whether I can change the point.id  like below code.
 for i in range ( 1, 1000):
     model.space.id = i
     model.space.pv_net_cf() 

Thank you for all the reply!  
Sincerely

**Reply by fumitoh** [reply]
A similar question was recently posted. Please refer to #97. The answer in discussion #97 explains how you can create a model for a subset of model points and delete it.

In the future, I would like to enhance modelx to allow for the deletion of individual spaces in exported models using the `del` statement, as demonstrated below:

```python
result = 0
for i in range(1, 1000):
    result += model.space[i].pv_net_cf()
    del model.space[i]
```

**Reply by happyych** [reply]
Thank you for your reply and I'm sorry that I missed the similar question you already answered.
I am looking forward to use the new "del" statement in the future. Thanks!
